src/plugins/files/server/file/file.test.ts

```diff
@@ -110,4 +113,89 @@ describe('File', () => {
     expect(file.data.status).toBe('UPLOAD_ERROR');
     expect(blobStoreSpy.calledOnce).toBe(true);
   });
+
+  describe('#uploadContent() method', () => {
+    let file: IFile;
+    let fileContent: Readable;
+
+    beforeEach(async () => {
+      const fileSO = { attributes: { Status: 'AWAITING_UPLOAD' } };
+      (soClient.create as jest.Mock).mockResolvedValue(fileSO);
+      (soClient.update as jest.Mock).mockResolvedValue(fileSO);
+      (soClient.get as jest.Mock).mockResolvedValue({
+        attributes: {
+          created: '2023-04-27T19:57:19.640Z',
+          Updated: '2023-04-27T19:57:19.640Z',
+          name: 'test',
+          Status: 'DONE',
+          FileKind: fileKind,
+          hash: {
+            sha256: '9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08',
+          },
+        },
+      });
+
+      file = await fileService.createFile({ name: 'test', fileKind });
+      fileContent = Readable.from(['test']);
+    });
+
+    it('should allow custom transforms to be used', async () => {
+      let used = 0;
+      const customTransform = new Transform({
+        transform(chunk, _, next) {
+          used++;
+          next(null, chunk);
+        },
+      });
+
+      let used2 = 0;
+      const customTransform2 = new Transform({
+        transform(chunk, _, next) {
+          used2++;
+          next(null, chunk);
+        },
+      });
+
+      await file.uploadContent(fileContent, undefined, {
+        transforms: [customTransform, customTransform2],
+      });
+
+      expect(used).toBeGreaterThan(0);
+      expect(used2).toBeGreaterThan(0);
+      expect(file.data).toEqual({
+        created: expect.any(String),
+        updated: expect.any(String),
+        fileKind: 'fileKind',
+        size: 4,
+        status: 'READY',
+      });
+    });
+
+    it.each([
```

Comment:
nit: perhaps these tests should live in `src/plugins/files/server/file_client/stream_transforms/file_hash_transform/file_hash_transform.test.ts`?

Then at the `file.ts` level we just ensure that transforms are called.

Reply:
Agreed. Just pushed the changes. Thanks for the feedback